x-pack/plugins/observability_solution/infra/server/routes/infra/lib/host/get_all_hosts.ts

```diff
@@ -5,78 +5,67 @@
+import { METADATA_AGGREGATION_NAME } from '../constants';
+import type { GetHostParameters } from '../types';
+import {
+  getFilterByIntegration,
+  getInventoryModelAggregations,
+  getValidDocumentsFilter,
+} from '../helpers/query';
 import { BasicMetricValueRT } from '../../../../lib/metrics/types';
 
-export const getAllHosts = async (
-  { infraMetricsClient, params }: GetHostsArgs,
-  hostNamesShortList: string[] = []
-) => {
-  const query = createQuery(params, hostNamesShortList);
-  const response = await infraMetricsClient.search(query);
-
-  const result = (response.aggregations?.nodes.buckets ?? [])
-    .sort((a, b) => {
-      const aValue = getMetricValue(a?.cpuV2) ?? 0;
-      const bValue = getMetricValue(b?.cpuV2) ?? 0;
-      return bValue - aValue;
-    })
-    .map((bucket) => {
-      const metadata = (bucket?.metadata.top ?? [])
-        .flatMap((top) => Object.entries(top.metrics))
-        .map(([key, value]) => ({
-          name: key as InfraAssetMetadataType,
-          value: typeof value === 'string' && value.trim().length === 0 ? null : value,
-        }));
-
-      const metrics = params.metrics.map((metric) => ({
-        name: metric.type,
-        value: metric.type in bucket ? getMetricValue(bucket[metric.type]) ?? 0 : null,
-      }));
-
-      return {
-        name: bucket?.key as string,
-        metadata,
-        metrics,
-      };
-    });
-
-  return result;
-};
-
-const createQuery = (params: GetInfraMetricsRequestBodyPayload, hostNamesShortList: string[]) => {
+export const getAllHosts = async ({
+  infraMetricsClient,
+  from,
+  to,
+  limit,
+  metrics,
+  hostNames,
+}: Pick<GetHostParameters, 'infraMetricsClient' | 'from' | 'to' | 'limit' | 'metrics'> & {
+  hostNames: string[];
+}) => {
   const metricAggregations = getInventoryModelAggregations(
-    params.type,
-    params.metrics.map((p) => p.type)
+    'host',
+    metrics.map((metric) => metric)
   );
 
-  return {
+  const response = await infraMetricsClient.search({
     allow_no_indices: true,
     ignore_unavailable: true,
     body: {
       size: 0,
       track_total_hits: false,
       query: {
         bool: {
-          filter: createFilters({
-            params,
-            hostNamesShortList,
-          }),
+          filter: [...termsQuery(HOST_NAME_FIELD, ...hostNames), ...rangeQuery(from, to)],
+          should: [...getValidDocumentsFilter()],
         },
       },
       aggs: {
-        nodes: {
+        // find hosts with metrics that are monitored by the system integration.
+        monitoredHosts: {
+          filter: getFilterByIntegration('system'),
+          aggs: {
+            names: {
+              terms: {
+                field: HOST_NAME_FIELD,
+                size: limit,
+                order: {
+                  _key: 'asc',
+                },
+              },
+            },
+          },
+        },
+        allHostsMetrics: {
```

Comment:
Nit maybe we can rename it to `allHostMetrics`, wdyt?
```suggestion
        allHostMetrics: {
```